R-package/src/install.libs.R

```diff
@@ -44,6 +44,14 @@ if (!use_precompile) {
     cmake_cmd <- paste0(cmake_cmd, " -DUSE_R35=ON ")
   }
 
+  if (Sys.info()['sysname'] == 'Darwin' & !(grepl('^gcc', Sys.getenv('CC', '')) & grepl('^g++'), Sys.getenv('CXX', ''))) {
```

Comment:
could you please add a comment above this block explaining its intent?

Also, could you please use `&&` so we get short-circuiting? If you've never seen that before in R, try this:

```
> FALSE & stop("error")
Error: error

> FALSE && stop("error")
[1] FALSE
```

Reply:
Thanks for a good example.